src/libDirectoryService/DirectoryService.cpp

```diff
@@ -507,3 +507,93 @@ std::string DirectoryService::GetActionString(Action action) const
         ? "Unknown"
         : ActionStrings.at(action);
 }
+
+uint8_t
+DirectoryService::CalculateNewDifficulty(const uint8_t& currentDifficulty)
+{
+    const int8_t MAX_ADJUST_STEP = 2;
+    int64_t requiredNodes = 0, powSubmissions = 0;
+    {
+        lock_guard<mutex> g(m_mutexAllPOW);
+        powSubmissions = m_allPoWs.size();
+
+        for (const auto& shard : m_shards)
+        {
+            requiredNodes += shard.size();
```

Comment:
Why is the required number of nodes, the current network size, shouldn't it be the sweet spot ??